eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Persistence\Tests\Legacy\FieldValue\Converter;
+
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\IO\UrlRedecoratorInterface;
+use eZ\Publish\Core\IO\Values\BinaryFile;
+use eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\ImageConverter;
+use eZ\Publish\Core\Persistence\Legacy\Content\StorageFieldValue;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\PhpUnit\ClockMock;
+
+final class ImageConverterTest extends TestCase
+{
+    /** @var \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\ImageConverter */
+    private $imageConverter;
+
+    /** @var \eZ\Publish\Core\IO\UrlRedecoratorInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $urlRedecorator;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $ioService;
+
+    protected function setUp(): void
+    {
+        $this->ioService = $this->createMock(IOServiceInterface::class);
+        $this->urlRedecorator = $this->createMock(UrlRedecoratorInterface::class);
+
+        $this->imageConverter = new ImageConverter(
+            $this->ioService,
+            $this->urlRedecorator
+        );
+    }
+
+    public function testToStorageValue(): void
+    {
+        ClockMock::register(ImageConverter::class);
+        ClockMock::withClockMock(true);
+
+        $pathToImg = __DIR__ . '/../_fixtures/ibexa_fav.png';
+        $dir = __DIR__ . '/../_fixtures';
+        $time = ClockMock::time();
+
+        $expectedXml = <<< XML
+<?xml version="1.0" encoding="utf-8"?>
+<ezimage serial_number="1" is_valid="1" filename="ibexa_fav.png"
+    suffix="png" basename="ibexa_fav" dirpath="{$dir}" url="{$pathToImg}"
+    original_filename="ibexa_fav.png" mime_type="image/png" width="100"
+    height="200" alternative_text="test" alias_key="1293033771" timestamp="{$time}">
+  <original attribute_id="1" attribute_version="1" attribute_language="eng-GB"/>
+  <information Height="200" Width="100" IsColor="1"/>
+  <additional_data focalPointX="50" focalPointY="100" author="John Smith"/>
```

Comment:
Are those data attributes/options dynamic (source system dependent and different based on a system)?
If so, then this XML should rather be something like:
```xml
<additional_data>
  <option key="focalPointX">50</option>
  <option key="focalPointY">100</option>
  <option key="author">John Smith</option>
</additional_data>
```
Dynamic values should never be neither XML tags nor attributes, so XSD schema can be derived easily.
Note that I took "option" from `buildAdditionalDataTag`, but it seems those are attributes? Maybe then it should be `attributes > attribute` instead?

Reply:
I redone it in https://github.com/ezsystems/ezplatform-kernel/pull/137/commits/1429f5b67ba405853fd73c3b0b8baf98f9affef2